### PR TITLE
Push new compiler binaries sequentially after each `amp-closure-compiler` PR is merged to `master`

### PR DESCRIPTION
**Pull request by rsimha**
**PR highlights:**

- Updates build code to always generate a new compiler binary for every change to the `amp-closure-compiler` repo
- Updates push workflow as follows:
    - Create a commit with the updated compiler binary for the current OS
    - Revert all other local changes (extraneous test files, intermediate build output, etc.)
    - Pull latest changes from `origin` and rebase local commit on them
    - Push all local commits to `origin`, including the newly built compiler binaries

### CODE REVIEW

File: tasks/push-binaries.js [hunk 1]
```diff
@@ -31,13 +31,23 @@ const platformOsMap = {
...
+  execOrDie(`git add ${compilerBinaryPaths}`);
   execOrDie(`git commit -m "Push new ${osName} compiler binaries"`);
+  execOrDie('git diff');
+  execOrDie('git status');
   execOrDie('git pull --rebase');
   execOrDie('git push');
-})();
+}
+
+main();
```
Comment (kristoferbaxter) [inline tasks/push-binaries.js:51]: Any reason to not use the existing immediate execution?

Comment (rsimha) [reply tasks/push-binaries.js:51]: GH actions didn't seem happy with it. I tried debugging to see if it was a typo, but decided to change the call to proceed. Happy to switch it back if you can spot the problem.

image

Comment (kristoferbaxter) [reply tasks/push-binaries.js:57]: Wild. No issue from my perspective to change.

Comment (rsimha) [reply tasks/push-binaries.js:47]: Yeah, mysterious. I'll investigate this in a follow up after this is merged.

**Comment by rsimha** [comment]
Last week's approach of updating the PR branch did not work because PRs get a read-only Github token. The good news though, is that I now have a fully working solution to push three sets of binaries from three parallel jobs such that the commits do not overwrite one another.

Here is a screenshot of a successful automatic upload to a feature branch on my personal fork. The first commit was what I pushed, the others were pushed by the three GH actions workflows.
image

I've updated this PR with the latest working (🤞 ) code, and will merge it to see if the upload happens to `master`. Once that happens, I can work on trying this out on `ampproject/amphtml` and follow up with a cleanup of the code in this repo.

**Comment by rsimha** [comment]
Yay! This worked.

image

Note: I had to disable some of the branch protection rules I had instituted for `master` (e.g. only reviewed PRs may be committed, etc.). Will have to revisit them in future to see if there's a way to protect `master` while allowing an action to auto-upload commits.

**Pull request merged**